lib/services/consumed/autocomplete.js

```diff
@@ -0,0 +1,49 @@
+/* @flow */
+import { CompositeDisposable, Disposable } from "atom";
+
+import type { Store } from "../../store";
+
+export const autocompleteProvider = {
+  disposables: new CompositeDisposable(),
+  watchEditor: (editor: atom$TextEditor, labels: Array<string>) => {
+    return;
+  }
+};
+
+export function addWatchEditor(
+  store: Store,
+  watchEditor: atom$AutocompleteWatchEditor
+) {
+  autocompleteProvider.watchEditor = watchEditor;
+  for (let kernel of store.runningKernels) {
+    for (let watch of kernel.watchesStore.watches) {
+      const disposable = watchEditor(this.editor, [
+        "default",
+        "workspace-center",
+        "symbol-provider"
+      ]);
+      if (disposable) {
+        watch.autocompleteDisposable = disposable;
+        kernel.watchesStore.autocompleteDisposables.add(disposable);
+      }
+    }
+    autocompleteProvider.disposables.add(
+      kernel.watchesStore.autocompleteDisposables
+    );
+  }
+  return new Disposable(this.disableAutocomplete.bind(this, store));
```

Comment:
Do we need to bind `this` and wouldn't `this.disableAutocomplete` be the same as `disableAutocomplete` since we are not inside of a `class`? 

In other words, would this work?

```js
return new Disposable(() => disableAutocomplete(store));
```

Reply:
I'll do some testing 😃

Reply:
oh wait yes this work. I had it like that originally, but I thought my way better. After some thought, I like your way better, cleaner and almost exactly the same.